server/session/event_log.go

```
@@ -252,45 +252,301 @@ func parseEventsFromReader(reader *bufio.Reader) (parsedEvents, error) {
+	ReachedEnd   bool
+}
+
+func readSegmentBackwardFile(path string, endOffset int64, chunkBytes int64, match func(Event) bool) (SegmentWindow, error) {
+	if chunkBytes <= 0 {
+		chunkBytes = activeTailReverseChunkBytes
+	}
+	fp, err := openRegularSessionFile(path, "events file")
+	if err != nil {
+		return SegmentWindow{}, fmt.Errorf("open events file: %w", err)
+	}
+	defer fp.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `fp.Close` is not checked

(errcheck)